src/celengine/dsorenderer.cpp

```diff
@@ -55,12 +55,11 @@ brightness(float avgAbsMag, float absMag, float appMag, float brightnessCorr, fl
 
 } // anonymous namespace
 
-DSORenderer::DSORenderer() :
-    ObjectRenderer<DeepSkyObject*, double>(DSO_OCTREE_ROOT_SIZE)
+DSORenderer::DSORenderer() : ObjectRenderer(DSO_OCTREE_ROOT_SIZE)
 {
 }
 
-void DSORenderer::process(DeepSkyObject* const &dso,
+void DSORenderer::process(const std::unique_ptr<DeepSkyObject>& dso, //NOSONAR
```

Comment:
why ref to uniqptr instead of raw ptr?

Reply:
Because that's how the base template class `OctreeProcessor<OBJ, PREC>` defines it. In the case of the DSO octree, `OBJ` is `std::unique_ptr<DeepSkyObject>` (previously `DeepSkyObject*`), and defines the pure virtual method `process` as taking a parameter of type `const OBJ&`, which ends up being `const std::unique_ptr<DeepSkyObject>&` (previously `DeepSkyObject* const&`).